app/scripts/lib/derived-keys.js

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * Derive relier-specific encryption keys from account master keys.
+ */
+
+'use strict';
+
+define([
+  'sjcl',
+  'lib/hkdf',
+  'lib/constants'
+], function (sjcl, hkdf, Constants) {
+
+  /**
+   * Given the account master keys, the user id and the relier id, generate
+   * the matching relier-specific derived keys.
+   */
+  function deriveRelierKeys(keys, uid, rid) {
+    var relierKeys = {};
+    return generateDerivedKey(keys.kA, 'kAr', uid, rid)
+      .then(function (kAr) {
+        relierKeys.kAr = kAr;
```

Comment:
Do we care that the key class name is the same as the property on the relierKeys object? Perhaps we should make 'kAr' a constant and then change this line to: `relierKeys[<constant_name>] = kAr;`. It is a little funny to have these magic strings in the code ('kAr' and 'kBr').